instruqt-tracks/network-infrastructure-automation/08-install-consul-terraform-sync/solve-workstation

```diff
@@ -6,7 +6,46 @@ echo ">> Installing Consul Terraform Sync"
 echo ">>"
 
 cd /root/terraform/consul-tf-sync
-terraform apply -auto-approve > /root/terraform/consul-tf-sync/terraform.out
+
+# Run terraform apply until it is successful
+n=0
+while /bin/true; do
+    n=$((n+1))
+    rm -f /root/terraform/consul-tf-sync/terraform.out
+    
+    echo "Terraform apply attempt $n"
+    terraform apply -auto-approve > /root/terraform/consul-tf-sync/terraform.out 2>&1
+    rc=$?
+    if [ "$rc" != "0" ]; then
+        echo "terraform apply failed with return code $rc"
+        echo "Output follows:"
+        cat /root/terraform/consul-tf-sync/terraform.out
+        echo "END OF OUTPUT"
+
+        # Does VM exist?
+        terraform state list "${RESOURCE_ADDR}"
+        ec=$?
+        if [ $ec -ne 0 ]; then
+            echo "${RESOURCE_ADDR} doesn't exist after Terraform is finished"
+            # Delete the half-created VM
+            az vm delete -g "${RG}" -n "${RESOURCE_NAME}" --yes
+            az disk delete -g "${RG}" -n "${RESOURCE_NAME}-disk" --yes
+            sleep 185	# THIS VALUE IS IMPORTANT, see IL-843
+            terraform destroy -auto-approve
+            TRY=$((TRY+1))
+            # we've already waited 185 seconds at least, so no need to sleep
+        else
+            echo "${RESOURCE_ADDR} provisioned"
+            touch .provisioned
```

Comment:
Does this `.provisioned` file get used anywhere?

Reply:
Im actually not sure, it was part of a helper function that was made for the setup script. @thomashashi